Azure/azure-cli
# [ARM] `az demployment group create`: Support deployment with bicepparam files

**Description**


The PR enables deployment of Bicep templates using Bicep parameter files.  

Bicep parameter files allow users to create parameters files using Bicep syntax and enable useful features such as type checks, code completion, etc (check Bicep issue [#9567](https://github.com/Azure/bicep/issues/9567#issue-1536999409) for details)

**Testing Guide**
```
az deployment group create \ 
-g <rg> \
-f .\bicepparam\storage_account_template.bicep \
-p .\bicepparam\storage_account_params.bicepparam  
```
Notes: 
- Bicep parameters file is currently behind an experimental feature flag so a `bicepconfig.json` with `paramsFiles`  set to `true` is needed in the directory with the `.bicepparam` file for compilation
- Integration tests in this PR are `LiveScenarioTest` because we need to do a Bicep CLI version check (meaning it has to be installed) before performing any `.bicepparam` related logic

**History Notes**


[ARM] `az deployment group create`: Support deployment with bicepparam files

---

This checklist is used to make sure that common guidelines for a pull request are followed.
